### PR TITLE
DB-7503: Spark 2.3 support - external table implementation

### PR DESCRIPTION
**Pull request by OlegMazurov**
Spark 2.3 support is enabled for one platform only: cdh5.13.0.
ColumnVector.java and ColumnarBatch.java, which have been relocated in Spark 2.3, are reinstated in the previous package, org.apache.spark.sql.execution.vectorized, and reimplemented using their counterparts from Spark 2.3. Thus, practically no source code changes are required in Splice code that uses those 2 classes.
Spark 2.3 code variants have been created in hbase_sql (orc-cds-2.3), splice-spark(spark-2.3) and  platform_it (spark2.3.0-cloudera3).
Performance of the new implementation has not been evaluated (a new JIRA will be filed to cover that).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1142,11 +1143,12 @@
...
                 <zookeeper.version>3.4.5-cdh5.13.0</zookeeper.version>
-                <spark.major.version>spark2.2</spark.major.version>
-                <spark.version>2.2.0.cloudera2</spark.version>
-                <kafka.version>0.10.2-kafka-2.2.0</kafka.version>
+                <spark.major.version>spark2.3</spark.major.version>
+                <spark.version>2.3.0.cloudera3</spark.version>
+                <kafka.version>0.10.0-kafka-2.1.0</kafka.version>
                 <splice.lib.dir>/opt/cloudera/parcels/SPLICEMACHINE/lib</splice.lib.dir>
-                <parquet.version>parquet-cloudera</parquet.version>
+                <parquet.version>orc-cds-2.3</parquet.version>
```
[the remark this answers is not in the document]
Comment (OlegMazurov) [reply pom.xml]: Previously we set kafka.version but never used it relying on a transitive dependency. That dependency was removed in Spark 2.3 so I had to add it in platform_it/pom.xml with the right version provided by Cloudera.

They name parquet-* had little to do with the parquet format as it turned out. ColumnVector borrowed the Dictionary class from parquet and Cloudera and other vendors used different packages for that class (that was "fixed" in Spark2.3  and now I need both parquet Dictionary and ParquetDictionary). Essentially, I need to provide Spark variance in hbase_sql for different implementations affecting our ORC reader. That variance is different from other two cases: spark.major.version in splice_spark and spark.version in platform_it. I kept the name, now a misnomer, but gave a more meaningful value, orc-cds-2.3.
We'll need to clean up that mess once we drop some older platforms we don't support any more.

**Comment by dgomezferro** [comment]
We have to be careful when merging this and the other PRs that modify version dependent classes in splice_spark, since those changes won't be applied to the Spark 2.3 classes.

**Comment by OlegMazurov** [comment]
Not ready for merge!
Spark Adapter tests fail after synch up with master (DB-7541)

**Comment by OlegMazurov** [comment]
Ready to be merged. Successful branch build for cdh5.13.0 using Spark 2.3: http://nexus.splicemachine.com:8080/job/spliceengine-branch/job/spliceengine-branch/1271/default/

**Pull request merged**